.github/workflows/publish.yml

```diff
@@ -1,21 +1,97 @@
-name: Publish Package to npm
+name: Publish
+
 on:
-  push:
-    tags:
-      - 'v*'
-  workflow_dispatch:
+  # When Release Pull Request is merged
+  pull_request:
+    branches:
+      - main
+    types: [closed]
+
+permissions:
+  contents: write
+  issues: write
+  pull-requests: write
 
 jobs:
-  build:
+  publish:
+    if: github.event.pull_request.merged == true
     runs-on: ubuntu-latest
+
     steps:
-      - uses: actions/checkout@v4
-      # Setup .npmrc file to publish to npm
-      - uses: actions/setup-node@v4
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Use Node.js 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Git Identity
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+          git remote set-url origin https://x-access-token:${GITHUB_TOKEN}@github.com/$GITHUB_REPOSITORY
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      # Define ${CURRENT_VERSION}
+      - name: Set Current Version
+        run: echo "CURRENT_VERSION=$(cat package.json | jq -r .version)" >> $GITHUB_ENV
+
+      - name: Tag Check
+        id: tag_check
+        run: |
+          GET_API_URL="https://api.github.com/repos/${GITHUB_REPOSITORY}/git/ref/tags/v${CURRENT_VERSION}"
+          http_status_code=$(curl -LI $GET_API_URL -o /dev/null -w '%{http_code}\n' -s \
+            -H "Authorization: token ${GITHUB_TOKEN}")
+          if [ "$http_status_code" -ne "404" ] ; then
+            echo "exists_tag=true" >> $GITHUB_OUTPUT
+          else
+            echo "exists_tag=false" >> $GITHUB_OUTPUT
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Publish
+        if: steps.tag_check.outputs.exists_tag == 'false'
```

Comment:
It seems this script is not working properly. Should this condition be set to `steps.tag_check.outputs.exists_tag == 'true'` ?
https://github.com/ubie-oss/ubie-icons/actions/runs/8985341603/job/24679125682

Reply:
it shuold be 'false'.

Because `exists_tag` is true if the version in the package.json and latest tag's version are aligned, and if so      we don't want to release it.

Reply:
That means the reason it's not working properly this time is because the tags and the version in the `package.json` don't match. At a glance, they seem to match though...